src/core/wee-network.c

```diff
@@ -554,11 +554,13 @@ network_pass_socks4proxy (struct t_proxy *proxy, int sock, const char *address,
     socks4.method = 1;
     socks4.port = htons (port);
     network_resolve (address, ip_addr, NULL);
-    socks4.address = inet_addr (ip_addr);
     strncpy (socks4.user, username, sizeof (socks4.user) - 1);
 
     free (username);
 
+    if (inet_pton (AF_INET, ip_addr, &socks4.address) != 1)
+        return 1;
```

Comment:
Last change to do: return value must be 0 here (error).

Reply:
Done.